spec/ParseGraphQLServer.spec.js

```diff
@@ -4875,6 +4875,181 @@ describe('ParseGraphQLServer', () => {
           expect(Date.parse(getResult.data.get.updatedAt)).not.toEqual(NaN);
         });
 
+        it('should support ACL', async () => {
+          const someClass = new Parse.Object('SomeClass');
+          await someClass.save();
+
+          const user = new Parse.User();
+          user.set('username', 'username');
+          user.set('password', 'password');
+          await user.signUp();
+
+          const user2 = new Parse.User();
+          user2.set('username', 'username2');
+          user2.set('password', 'password2');
+          await user2.signUp();
+
+          const roleACL = new Parse.ACL();
+          roleACL.setPublicReadAccess(true);
+
+          const role = new Parse.Role('aRole', roleACL);
+          await role.save();
+
+          const role2 = new Parse.Role('aRole2', roleACL);
+          await role2.save();
+
+          await parseGraphQLServer.parseGraphQLSchema.databaseController.schemaCache.clear();
+
+          const {
+            data: { createSomeClass },
+          } = await apolloClient.mutate({
+            mutation: gql`
+              mutation Create($fields: CreateSomeClassFieldsInput) {
+                createSomeClass(fields: $fields) {
+                  id
+                  ACL {
+                    users {
+                      userId
+                      read
+                      write
+                    }
+                    roles {
+                      roleName
+                      read
+                      write
+                    }
+                    public {
+                      read
+                      write
+                    }
+                  }
+                }
+              }
+            `,
+            variables: {
+              fields: {
+                ACL: {
+                  users: [
+                    { userId: user.id, read: true, write: true },
+                    { userId: user2.id },
+                  ],
+                  roles: [
+                    { roleName: 'aRole', read: true },
+                    { roleName: 'aRole2' },
+                  ],
+                  public: { read: true, write: true },
+                },
+              },
+            },
+          });
+
+          const expectedCreateACL = {
+            __typename: 'ACL',
+            users: [
+              {
+                userId: user.id,
+                read: true,
+                write: true,
+                __typename: 'UserACL',
+              },
+              {
+                userId: user2.id,
+                read: true,
+                write: true,
+                __typename: 'UserACL',
+              },
+            ],
+            roles: [
+              {
+                roleName: 'aRole',
+                read: true,
+                write: false,
```

Comment:
why?

Reply:
Because if you only explicitly provide `read` to Parse it seems that `write` become false.
Maybe i'm wrong ?

Reply:
Since we've decided to make read and write defaulting to true, I think we should make it true in any case. Otherwise it will be an unexpected behavior of the GraphQL API.